ffi/derive/src/bindgen.rs

```diff
@@ -175,63 +221,53 @@ fn gen_output_assignment_stmts(fn_descriptor: &FnDescriptor) -> Vec<syn::Stmt> {
     let mut stmts = vec![];
 
     if let Some(output_arg) = &fn_descriptor.output_arg {
-        let output_arg_name = &output_arg.ffi_name;
+        let output_arg_name = &output_arg.name;
 
         if output_arg.is_slice_ref_mut() {
             let (slice_len_arg_name, slice_elems_arg_name) = (
-                gen_slice_len_arg_name(output_arg),
+                gen_slice_len_arg_name(&output_arg.name),
                 gen_slice_elems_arg_name(output_arg),
             );
 
             stmts.push(parse_quote! {{
-                let #output_arg_name = core::slice::from_raw_parts_mut(#output_arg_name, #slice_len_arg_name);
-
-                #slice_elems_arg_name.write(method_res.len());
-                for (i, elem) in method_res.take(#slice_len_arg_name).enumerate() {
-                    #output_arg_name[i] = elem;
+                #slice_elems_arg_name.write(#output_arg_name.len());
+                for (i, elem) in #output_arg_name.take(#slice_len_arg_name).enumerate() {
+                    let offset = i.try_into().expect("allocation too large");
```

Comment:
I could have used:
```suggestion
                    let offset = i as isize;
```

relevant [text](https://doc.rust-lang.org/std/primitive.pointer.html#method.offset)

Reply:
it could also be converted into an explicit error, instead of panicking

Reply:
This will fail silently, that wualifies as undefined behaviour (technically just a truncation error). However the truncation error only occurs if somewhere else we also have UB. To me this cast seems reasonable, but I would link to the text inside the code.

Reply:
added the link in the code. I've left it as explicit conversion, not a cast. Maybe the compiler would be smart enough to optimize it out on some platforms